File: Assets/Greco3D/scripts/task/Trial.js
#pragma strict
import UnityEngine.UI;

var numStr : String;
var timer : float;
var gui_on : boolean;
private var background : Background;
private var runbreak : boolean = true;
public var taskText : Text; 

function Start(){
background = GetComponent("Background");
}

function OnGUI(){
	if(gui_on){
    var  style : GUIStyle = new GUIStyle();
    var buttonWidth : int = 50;
	var buttonHeight : int = 20;
	var rect = new Rect();
	style.normal.textColor = Color.white;
	style.fontSize = 75;
	style.alignment = TextAnchor.MiddleCenter;
    rect =Rect((Screen.width /2)-(buttonWidth/2),(Screen.height/2) - (buttonHeight/2),buttonWidth,buttonHeight);
	GUI.Label(rect,numStr,style);
	}
}

function StartCountDown(){
	var timer : float = 3.5;	
	gui_on = true;
	background.BackGroundOn();


    while(timer > 0){
		timer -= Time.deltaTime;

	 	if(timer < 0.5){
	 		numStr ="";   
		}
	    else if(timer > .5){
	    	numStr =""+ Mathf.Round(timer);
	    }
		yield;
	}
}


function StartITI(){
	numStr ="+";
	print(numStr);
	gui_on = true;
	background.BackGroundOn();
}

function StopITI(){
	gui_on = false;
	background.BackGroundOff();
	numStr ="";
}

function RunBreak(){
numStr = "Press SPACE to continue";
gui_on = true;
background.BackGroundOn();
runbreak = true;
yield StartCoroutine(WaitForSpace());
numStr ="";
}

function WaitForSpace(){
	while(runbreak){
		if(Input.GetKeyUp(KeyCode.Space)){
			print("Spacebar pressed");
			runbreak = false;
		}
		yield;
	}
}

function TaskOver(){
numStr = "Task over";
gui_on = true;
background.BackGroundOn();
runbreak = true;
yield StartCoroutine(WaitForSpace());
numStr ="";
}

File: Assets/Greco3D/Trial.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class Trial : MonoBehaviour {

    public string numStr;
    public float timer;
    private Background background;
    private bool runBreak;
    public Text taskText;
    public Text accuracy;

    private void Start()
    {
        background = GetComponent(typeof(Background)) as Background;
        taskText.text = "";
        accuracy.text = "";
    }

    private void TextOn()
    {
         taskText.text = numStr;
    }

    public void TextOff()
    {
        taskText.text = "";
    }


    public IEnumerator StartCountDown()
    {
        float timer = 3.5f;
        //guiOn = true;
        TextOn();
        background.BackGroundOn();

        while (timer > 0f)
        {
            timer -= Time.deltaTime;

            if (timer < 0.5f)
            {
                taskText.text = "";
            }
            else if (timer > .5f)
            {
                taskText.text = "" + Mathf.Round(timer);
            }
            yield return null;
        }
    }

    public IEnumerator StartITI()
    {
        numStr = "+";
        background.BackGroundOn();
        yield return new WaitForSeconds(.3f);
        //guiOn = true;
        TextOn();
    }

    public IEnumerator StopITI()
    {
        //guiOn = false;
        TextOff();

        yield return new WaitForSeconds(.3f);
        background.BackGroundOff();
    }

    public IEnumerator RunBreak()
    {

        numStr = "Press SPACE to continue";
        //guiOn = true;
        TextOn();
        background.BackGroundOn();
        runBreak = true;
        accuracy.text = Manager.experiment.accPerc;
        yield return StartCoroutine(WaitForSpace());
        accuracy.text = "";
        numStr = "";
    }
    IEnumerator WaitForSpace()
    {
        while (runBreak)
        {
            if (Input.GetKeyUp(KeyCode.Space))
            {
                Debug.Log("Spacebar pressed");
                runBreak = false;
            }
            yield return null;
        }
    }

}
